### DISCUSSION TITLE
[Suggestion] Remove/Alter Memory Bookmarks "Description Override" feature

### DISCUSSION THREAD

**Discussion by cdxemu** [RetroAchievements/RAIntegration#1017] [Ideas]
I'd like to get a sense of whether any Devs use the Memory Bookmarks window's feature to override the "Bookmark Description". (We could survey this on Discord as well.)

I personally don't find it useful, but I often find myself mistakenly editing the descriptions while thinking I am editing the Code Notes, only to later find that the Code Notes weren't updated. I've also overheard other Jr Devs mention that they did something similar. Furthermore, the Memory Bookmarks window seems more convenient for editing multiple code notes at once compared to Memory Inspector, as long as they aren't multi-line formatted.

If it's not being used for its intended purpose, I'd suggest removing the feature and have the Memory Bookmarks "Description" field edit the Code Notes instead. On the other hand, if any Devs are using the feature, I'd suggest instead adding highlighting to indicate that text is an override that doesn't match the current Code Note.

**Comment by redwizard42** [comment]
I use it sometimes, especially when my note is necessarily long for detail-purposes, but I need something more compact in bookmarks so that I can tell at a glance what it means.

**Comment by cdxemu** [comment]
I can see that. I'd imagine it's most useful once your code notes are already written/stable. I wouldn't push to remove the feature if it's being used, so I'd suggest the highlighting (bold text?) instead.

The issue I described above mainly happens during memory digging while I am mass-adding memory addresses with blank Code Notes, finding the "correct" one, and then adding its Code Note. That's when I tend to accidentally use the Bookmarks window instead of Inspector.

**Comment by Jamiras** [comment]
The idea behind the feature is so you can label potential addresses until you know for certain what they do, so you are using it correctly. Once you've decided that the address is what you've labelled it, then you promote it to a note. 

Now that we have local notes, they could be more tightly integrated, where the unconfirmed description could be handled by a local note.

**Reply by cdxemu** [reply]
Hmm, that is surprising. I had thought the intended purpose was more like what TheMysticalOne described. I believe I got that impression from how it is described in the RADocs: "The bookmarks can be renamed by clicking their description. This does not update the code note for the associated address." If it is intended for unconfirmed notes, that doesn't seem as useful (since that is now covered by local notes, as you explained).

But the issue for me is not the usefulness of the feature (or lack thereof), but that it's too easy to absentmindedly use it when you are actually intending to edit the code note. So assuming we want to keep the feature, I like the idea of highlighting the text in bold if it doesn't match the current code note, or maybe asking the user to confirm that they are only overriding the Bookmark text.

**Reply by TeddyWestside22** [reply]
I use them as Jamiras described, by noting addresses that I'm not sure of and only putting them in as code notes once I have fully confirmed what the memory is doing. Some games I will only have bookmarks after several hours of playing and no code notes, then convert them to code notes. If the feature wasn't there, it would just make it confusing to me "was this note actually confirmed or did I simply write it down to keep track?", things like that. Or I'd have to write "unconfirmed" then go back and fix it once I'm sure. The bookmark system makes it easier to do and see at a glance in my opinion.

I don't have an issue with overriding and thinking it's a code note, because I always do them before I make the code note and monitor how the memory changes so I'm sure.
